wait/http_test.go

```diff
@@ -55,7 +55,7 @@ func ExampleHTTPStrategy_WithBasicAuth() {
 	req := testcontainers.ContainerRequest{
 		Image:        "gogs/gogs:0.11.91",
 		ExposedPorts: []string{"3000/tcp"},
-		WaitingFor:   wait.ForHTTP("/").WithPort("3000/tcp").WithBasicAuth("username", "password"),
+		WaitingFor:   wait.ForHTTP("/").WithBasicAuth("username", "password"),
```

Comment:
I would keep the `WithPort` part for the docs sections, as user could still want to see it in the code snippets regarding HTTP wait.

I'd add a new section in the docs, e.g. `Match an HTTP method with Port` keeping the old snippet as is, and in the current `Match an HTTP method`, I'd link it to the test code actually not using the `WithPort`.

If you are not sure on this, I can send a follow-up PR refining the docs, whatever you prefer is fine.

Reply:
Ah OK I added back an example with `WithPort` - I added an unrelated port to the `ExposedPorts` to illustrate where it would be useful